lib/services/cache/cache_persist.js

```diff
@@ -104,45 +95,43 @@ function get(key, opts, callback) {
   }
 
   if (!opts) opts = {};
-
-  var cached = this.__cache[key];
-
+  let cached = this.__cache[key];
   if (cached != null) return this.__tryCallback(callback, cached.data, null, true);
-
   if (this.__synced) return this.__tryCallback(callback, null, null); //we are in sync with the db, null value
 
-  var _this = this;
-
-  _this.__getData(key, function (e, item) {
+  this.__getData(key, (e, item) => {
 
     if (e) return callback(e);
 
-    if (item) return _this.__tryCallback(callback, item.data, null, true);
-
-    if (opts.retrieveMethod)
-      return opts.retrieveMethod.call(opts.retrieveMethod, function (e, result) {
+    if (item) {
+      return this.set(key, item.data.data, {
+        ttl: item.data.ttl,
+        noPersist: true
+      }, (e) => {
+        this.__tryCallback(callback, item, e, true);
+      });
+    }
 
+    if (opts.retrieveMethod){
+      return opts.retrieveMethod.call(opts.retrieveMethod, (e, result) => {
         if (e) return callback(e);
-
         // -1 and 0 are perfectly viable things to cache
-        if (result == null || result == undefined) return _this.__tryCallback(callback, null, null);
-
-        _this.set(key, result, opts, function (e) {
-          return _this.__tryCallback(callback, result, e, true);
+        if (result == null || result == undefined) return this.__tryCallback(callback, null, null);

```

Comment:
again

Reply:
ag, old code that must have moved - will sort out.